api/nodes/serializers.py

```diff
@@ -1721,3 +1726,116 @@ def enable_or_disable_addon(self, obj, should_enable, addon_name, auth):
         if type(addon) == bool:
             addon = None
         return addon
+
+
+class NodeGroupsSerializer(JSONAPISerializer):
+    filterable_fields = frozenset([
+        'name',
+        'permission',
+        'date_created',
+    ])
+
+    writeable_method_fields = frozenset([
+        'permission',
+    ])
+
+    id = CompoundIDField(source='_id', read_only=True)
+    type = TypeField()
+    permission = ser.SerializerMethodField()
+    name = ser.CharField(read_only=True)
+    date_created = VersionedDateTimeField(source='created', read_only=True)
+    date_modified = VersionedDateTimeField(source='modified', read_only=True)
+
+    osf_groups = RelationshipField(
```

Comment:
I'm not sure the `osf_` prefix is necessary for client-facing fields. We use "OSF groups" internally to disambiguate with Django groups, but this ambiguity shouldn't exist for clients.

Reply:
good point.